Sources/Plasma/FeatureLib/pfPython/pyImageLibModGlue.cpp

```diff
@@ -0,0 +1,186 @@
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+*==LICENSE==*/
+
+
+#include <Python.h>
+#include "pyKey.h"
+
+#include "pyImageLibMod.h"
+#include "pyImage.h"
+#include "plGImage/plMipmap.h"
+
+// glue functions
+PYTHON_CLASS_DEFINITION(ptImageLibMod, pyImageLibMod);
+
+PYTHON_DEFAULT_NEW_DEFINITION(ptImageLibMod, pyImageLibMod)
+PYTHON_DEFAULT_DEALLOC_DEFINITION(ptImageLibMod)
+
+PYTHON_INIT_DEFINITION(ptImageLibMod, args, keywords)
+{
+    PyObject* keyObj = nullptr;
+    if (!PyArg_ParseTuple(args, "O", &keyObj))
+    {
+        PyErr_SetString(PyExc_TypeError, "__init__ expects a ptKey");
+        PYTHON_RETURN_INIT_ERROR;
+    }
+    if (!pyKey::Check(keyObj))
+    {
+        PyErr_SetString(PyExc_TypeError, "__init__ expects a ptKey");
+        PYTHON_RETURN_INIT_ERROR;
+    }
+    pyKey* key = pyKey::ConvertFrom(keyObj);
+    self->fThis->setKey(*key);
+    PYTHON_RETURN_INIT_OK;
+}
+
+PYTHON_RICH_COMPARE_DEFINITION(ptImageLibMod, obj1, obj2, compareType)
+{
+    if ((obj1 == Py_None) || (obj2 == Py_None) || !pyImageLibMod::Check(obj1) || !pyImageLibMod::Check(obj2))
+    {
+        // if they aren't the same type, they don't match, obviously (we also never equal none)
+        if (compareType == Py_EQ)
+            PYTHON_RCOMPARE_FALSE;
+        else if (compareType == Py_NE)
+            PYTHON_RCOMPARE_TRUE;
+        else
+        {
+            PyErr_SetString(PyExc_NotImplementedError, "invalid comparison for a ptImageLibMod object");
+            PYTHON_RCOMPARE_ERROR;
+        }
+    }
+    pyImageLibMod* ilm1 = pyImageLibMod::ConvertFrom(obj1);
+    pyImageLibMod* ilm2 = pyImageLibMod::ConvertFrom(obj2);
+    if (compareType == Py_EQ)
+    {
+        if ((*ilm1) == (*ilm2))
+            PYTHON_RCOMPARE_TRUE;
+        PYTHON_RCOMPARE_FALSE;
+    }
+    else if (compareType == Py_NE)
+    {
+        if ((*ilm1) != (*ilm2))
+            PYTHON_RCOMPARE_TRUE;
+        PYTHON_RCOMPARE_FALSE;
+    }
+    PyErr_SetString(PyExc_NotImplementedError, "invalid comparison for a ptImage object");
+    PYTHON_RCOMPARE_ERROR;
+}
```

Comment:
I would be tempted to leave this out so one of the more standard type macros could be used.

Reply:
This is all boilerplate, the same as is present in other py*glue.

Reply:
I'm not sure which macros you mean.  As far as I can tell this is consistent with other Python class glue.